pkg/snapshot/restorer/restorer_test.go

```diff
@@ -665,290 +665,338 @@ var _ = Describe("Running Restorer", func() {
+			Expect(err).ShouldNot(HaveOccurred())
+			cancel()
+
+			//Add data into etcd
+			start = stop
+			stop = stop + 10
+			utils.PopulateEtcd(testCtx, logger, ep, start, stop, resp)
+			Expect(resp.Err).ShouldNot(HaveOccurred())
+			//Take delta snapshot
+			ctx, cancel = context.WithTimeout(testCtx, time.Duration(2*time.Second))
+			err = utils.RunSnapshotter(logger, snapstoreConfig, deltaSnapshotPeriod, ep, ctx.Done(), false, compressionConfig)
+			Expect(err).ShouldNot(HaveOccurred())
+			cancel()
 
-				//Add data into etcd
-				utils.PopulateEtcd(testCtx, logger, endpoints, 0, keyTo, resp)
-				Expect(resp.Err).ShouldNot(HaveOccurred())
-				//Take delta snapshot
-				ctx, cancel = context.WithTimeout(testCtx, time.Duration(2*time.Second))
-				snapstoreConfig = brtypes.SnapstoreConfig{Container: snapstoreDir, Provider: "Local", Prefix: "v2"}
-				err = utils.RunSnapshotter(logger, snapstoreConfig, deltaSnapshotPeriod, endpoints, ctx.Done(), false, compressionConfig)
-				Expect(err).ShouldNot(HaveOccurred())
-				cancel()
+			baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+			Expect(err).ShouldNot(HaveOccurred())
 
-				// remove the member dir
-				err = os.RemoveAll(memberPath)
-				Expect(err).ShouldNot(HaveOccurred())
+			rstr = NewRestorer(store, logger)
 
-				baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
-				Expect(err).ShouldNot(HaveOccurred())
+			restoreOpts := brtypes.RestoreOptions{
+				Config:        restorationConfig,
+				BaseSnapshot:  baseSnapshot,
+				DeltaSnapList: deltaSnapList,
+				ClusterURLs:   clusterUrlsMap,
+				PeerURLs:      peerUrls,
+			}
 
-				rstr = NewRestorer(store, logger)
+			//Restore
 
-				restoreOpts := brtypes.RestoreOptions{
-					Config:        restorationConfig,
-					BaseSnapshot:  baseSnapshot,
-					DeltaSnapList: deltaSnapList,
-					ClusterURLs:   clusterUrlsMap,
-					PeerURLs:      peerUrls,
-				}
+			// remove the member dir
+			err = os.RemoveAll(path.Join(emDir, "member"))
+			Expect(err).ShouldNot(HaveOccurred())
 
-				//Restore
-				err = rstr.RestoreAndStopEtcd(restoreOpts)
-				Expect(err).ShouldNot(HaveOccurred())
-				err = utils.CheckDataConsistency(testCtx, restoreOpts.Config.RestoreDataDir, keyTo, logger)
-				Expect(err).ShouldNot(HaveOccurred())
-			})
+			err = rstr.RestoreAndStopEtcd(restoreOpts)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = utils.CheckDataConsistency(testCtx, restoreOpts.Config.RestoreDataDir, stop, logger)
+			Expect(err).ShouldNot(HaveOccurred())
 		})
+	})
 
-		Context("With first few snapshots in v1 dir and some more incr snapshots are in v2 dir", func() {
-			It("should restore from v1 dir and v2 dir", func() {
-				memberPath := path.Join(etcdDir, "member")
+	Context("With first few snapshots in v1 dir and some more incr snapshots are in v2 dir", func() {
+		It("should restore from v1 dir and v2 dir", func() {
```

Comment:
Shouldn't you also be taking some v2 snapshots in this spec? I see that only v1 snapstore is being populated before checking for restoration.